src/compiler/csharp_generator.cc

```diff
@@ -382,6 +382,10 @@ void GenerateServerClass(Printer* out, const ServiceDescriptor* service) {
       "/// <summary>Base class for server-side implementations of "
       "$servicename$</summary>\n",
       "servicename", GetServiceClassName(service));
+  out->Print(
+      "[grpc::BindService(typeof($classname$), "
+      "nameof($classname$.BindService))]\n",
```

Comment:
> but we should probably use a fully qualified name to eliminate potential name collisions.

I thought about that but there are other places in the generated service that reference the service class name without fully qualifying it.

Reply:
> nameof() is a C# 6 concept and as of now, the generated code does not require C# 6. I know C# 6 is pretty old already, but we can easily use a string "BindService" instead of nameof() and remove the concern.

Ok!